File: crypto-casino-backend/src/main/java/hu/bme/aut/crypto_casino_backend/dto/game/GameHistoryResponse.java
package hu.bme.aut.crypto_casino_backend.dto.game;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameHistoryResponse {

  private Long spinId;

  private int[] reels;

  private BigDecimal betAmount;

  private BigDecimal winAmount;

  private boolean isWin;

  private LocalDateTime timestamp;

}

File: crypto_casino_android/app/src/main/java/hu/bme/aut/crypto_casino_android/data/model/slot/GameHistoryResponse.kt
package hu.bme.aut.crypto_casino_android.data.model.slot

import java.math.BigDecimal
import java.time.LocalDateTime

data class GameHistoryResponse(
    val id: Long,
    val spinId: Long,
    val gameType: String,
    val betAmount: BigDecimal,
    val winAmount: BigDecimal,
    val reels: List<Int>?,
    val timestamp: LocalDateTime
)
